### PR TITLE
refactor: return initial exception and check if it's user error

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Refactor sqllab to return initial exception and use the status code to indicate whether this error should be logged as an exception. We are going to log from the `handle_api_exception` decorator instead of the lower level.

https://github.com/apache/superset/blob/9edb581f103d3e29b40d1c814cbe935a3d36f486/superset/views/base.py#L202

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/sqllab/command.py [hunk 1]
```diff
@@ -122,25 +110,7 @@ def run(  # pylint: disable=too-many-statements,useless-suppression
...
-                            message=ex.message,
-                            error_type=ex.error_type,
-                            level=ErrorLevel.ERROR,
-                        )
-                    ]
-                ) from ex
-            raise ex
         except Exception as ex:
-            query_id = query.id if query else None
-            logger.exception("Query %d: %s", query_id, type(ex))
```
Comment (pkdotson) [inline superset/sqllab/command.py:143]: would it be useful to still log query id here or no?

Comment (hughhhh) [reply superset/sqllab/command.py:143]: I don't think having the id helps us debug the issue at all tbh, with that information we can't do anything else.

the important thing is that we log the exception here (https://github.com/apache/superset/blob/9edb581f103d3e29b40d1c814cbe935a3d36f486/superset/views/base.py#L202) once it gets bubbled up

Comment (pkdotson) [reply superset/sqllab/command.py:143]: got it!

**Review by pkdotson — APPROVED** [review]
lgtm!

**Review by AAfghahi — APPROVED** [review]

**Review by eschutho — APPROVED** [review]

**Comment by eschutho** [comment]
@betodealmeida @john-bodley Here are more examples of us refactoring to bubble up errors to the flask or celery level.

**Pull request merged**